testsuite/features/step_definitions/command_steps.rb

```diff
@@ -117,6 +117,49 @@
 end
 
 # Channels
+When(/^I prepare a channel clone for strict mode testing$/) do
+  get_target('server').run('cp -r /srv/www/htdocs/pub/TestRepoRpmUpdates /srv/www/htdocs/pub/TestRepoRpmUpdates_STRICT_TEST')
+  get_target('server').run('rm -rf /srv/www/htdocs/pub/TestRepoRpmUpdates_STRICT_TEST/repodata')
+  %w[i586 src x86_64].each do |folder|
+    get_target('server').run("rm -f /srv/www/htdocs/pub/TestRepoRpmUpdates_STRICT_TEST/#{folder}/rute-dummy-2.0-1.2.*.rpm")
+  end
+  get_target('server').run('createrepo_c /srv/www/htdocs/pub/TestRepoRpmUpdates_STRICT_TEST')
+  get_target('server').run('gzip -dc /srv/www/htdocs/pub/TestRepoRpmUpdates/repodata/*-updateinfo.xml.gz > /tmp/updateinfo.xml')
+  get_target('server').run('modifyrepo_c --verbose --mdtype updateinfo /tmp/updateinfo.xml /srv/www/htdocs/pub/TestRepoRpmUpdates_STRICT_TEST/repodata')
+end
+
+Given(/^I am logged into the API$/) do
+  server_node = get_target('server')
+  api_url = "https://#{server_node.public_ip}/rhn/manager/api/auth/login"
+  response = get_target('server').run("curl -H 'Content-Type: application/json' -d '{'login': 'admin', 'password': 'admin'}' -i #{api_url}")
+  raise 'Failed to login to the API' unless response.code == 200
+end
+
+$package_amount = nil
```

Comment:
Please don't introduce code in between step definitions, in those files, we expect all wrapped into the step definitions and we can't rely in how many times this ruby file has been loaded.

Reply:
removed, using `add_context` and `set_context` now